owner-onboarding-server/src/ov_management/ov_reply.rs

```diff
@@ -0,0 +1,178 @@
+
+
+
+use std::{usize, fs::{self, File}, io::Read};
+use fdo_data_formats::{ownershipvoucher::OwnershipVoucher, Serializable};
+use openssl::{pkey::{Private, PKey}};
+use serde::{Serialize, Deserialize};
+use warp::{hyper::StatusCode, http::HeaderValue, Buf, reply::Json};
+
+const ACCEPTED_CONTENT_TYPE:[&str; 2] = ["application/x-pem-file","application/cbor"];
+#[derive(Clone, Copy)]
+#[derive(Serialize,Deserialize)]
+enum ErrorCode {
+    IncompleteVoucher,
+    ParseError,
+    InvalidNumberOfVouchers,
+    UnwonedVoucher,
+    RequireContentTypeHeader,
+    RequireVoucherCountHeader,
+    NoOwnershipVoucherFound,
+    UnknownDevice,
+}
+
+#[derive(Serialize,Deserialize)]
+struct ErrorMsg<T:Serialize> {
+    error_code: Option<ErrorCode>,
+    error_details : Option<ErrorDetails<T>>
+}
+#[derive(Serialize,Deserialize)]
+struct ErrorDetails<T:Serialize >  {
+    key : Option<ErrorCode>,
+    value: Option<T>
+}
+
+trait Response<T> {
+    fn generate(e: ErrorCode,value: T ) -> Json;
+}
+
+impl<T> Response<T> for ErrorMsg<T> where T: Serialize {
+    fn generate(e: ErrorCode,value: T) -> Json {
+          let e=     ErrorMsg {
+                error_code:  Some(e),
+                error_details: Some(ErrorDetails{
+                    key : Some(e),
+                    value: Some(value),
+                })
+            };
+        warp::reply::json(&e)
+    }
+}
+
+
+
+pub(crate) async fn add_ov(
+    count: HeaderValue,
+    content_type: HeaderValue, 
+    body: impl Buf,
+    store_path: String,
+    owner_key: PKey<Private>,
+) -> Result<impl warp::Reply,warp::Rejection> 
+{
+    // check for ov type
+    let content_type = content_type.to_str().unwrap();
+    let ovs;
+    match content_type {
+        "application/x-pem-file" => ovs= OwnershipVoucher::many_from_pem(body.chunk()),
```

Comment:
This doesn't implement the `application/cbor` type.

Reply:
Added the functionality